website/docs/reference/project-configs/query-comment.md

```diff
@@ -0,0 +1,270 @@
+
+<File name='dbt_project.yml'>
+
+```yml
+
+query-comment: string
+
+```
+
+</File>
+
+
+### Dictionary syntax
+The `query-comment` configuration also accepts a dictionary input, with the following keys:
+  * `comment` (optional, see above for default): The string to be injected to a query as a comment.
+  * `append` (optional, default=`false`): Whether a comment should be appended (added to the bottom of a query) or not (i.e. added to the top of a comment). By default, comments are added to the top of queries (i.e. `append: false`).
+
+This syntax is useful on databases like Snowflake which [remove leading SQL comments](https://docs.snowflake.com/en/release-notes/2017-04.html#queries-leading-comments-removed-during-execution).
+
+<File name='dbt_project.yml'>
+
+```yml
+
+query-comment:
+  comment: string
+  append: true | false
+
+```
+
+</File>
+
+## Examples
+
+### Inject a static comment
+The following example injects a comment that reads `/* executed by dbt */` into the header of the SQL queries that dbt runs.
+
+<File name='dbt_project.yml'>
+
+```yml
+query-comment: "executed by dbt"
+
+```
+
+</File>
+
+**Example output:**
+
+```sql
+/* executed by dbt */
+
+select ...
+```
+
+### Disable query comments
+
+<File name='dbt_project.yml'>
+
+```yml
+query-comment:
+
+```
+
+</File>
+
+Or:
+
+<File name='dbt_project.yml'>
+
+```yml
+query-comment: null
+
+```
+
+</File>
+
+### Inject a dynamic comment
+The following example injects a comment that varies based on the configured `user` specified in the active dbt target.
+
+<File name='dbt_project.yml'>
+
+```yml
+query-comment: "run by {{ target.user }} in dbt"
+
+```
+
+</File>
+
+**Example output:**
+
+```sql
+/* run by drew in dbt */
+
+select ...
+```
+
+### Append the default comment
+The following example uses the dictionary syntax to append (rather than prepend) the default comment.
+
+Note that the `comment:` field is omitted to allow the default to be appended.
+
+<File name='dbt_project.yml'>
+
+```yaml
+
+query-comment:
+  append: True
+```
+
+</File>
+
+**Example output:**
+
+```sql
+select ...
+/* {"app": "dbt", "dbt_version": "0.16.`0rc2`", "profile_name": "debug", "target_name": "dev", "node_id": "model.dbt2.my_model"} */
+;
+```
+
+### Append a custom comment
+The following example uses the dictionary syntax to append (rather than prepend) a comment that varies based on the configured `user` specified in the active dbt target.
+
+<File name='dbt_project.yml'>
+
+```yaml
+
+query-comment:
+  comment: "run by {{ target.user }} in dbt"
+  append: True
+```
+
+</File>
+
+**Example output:**
+
+```sql
+select ...
+/* run by drew in dbt */
+;
+```
+
+
+
+### Intermediate: Use a macro to generate a comment
+
+The `query-comment` config can reference macros in your dbt project. Simply create a macro with any name (`query_comment` is a good start!) in your `macros` directory, like so:
+
+<File name='macros/query_comment.sql'>
+
+```jinja2
+
+{% macro query_comment(node) %}
```

Comment:
@drewbanin — in the original docs, this had:
```
{% macro my_comment() %}

```
Whereas the more advanced example had an argument for `node`.

Do we need to define `node` as an argument, or is it optional?